api/v1alpha1/qperf_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// QperfPort is the TCP port where the qperf server and client listens
+const QperfPort = 19765
+
+// QperfConfigurationSpec contains configuration parameters
+// with scheduling options for the both the qperf client
+// and server instances.
+type QperfConfigurationSpec struct {
+	PodConfigurationSpec `json:",inline"`
+
+	// HostNetwork requested for the qperf pod, if enabled the
+	// hosts network namespace is used. Default to false.
+	// +optional
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+}
+
+// QperfSpec defines the Qperf Benchmark Stone which
+// consist of server deployment with service definition
+// and client pod.
+type QperfSpec struct {
+	// Image defines the qperf docker image used for the benchmark
+	Image ImageSpec `json:"image"`
+
+	// Options are options for the qperf binary
+	// +optional
+	Options string `json:"options,omitempty"`
```

Comment:
Having the Options in top level preventing to specify different parameters for server and client.
Please consider moving this to QperfConfigurationSpec.

Reply:
I did not find any configuration parameter which would be usable on the server node. But if it is needed then we can pass it to the server spec.